Backend/controllers/CaseInvestigationController.js

```diff
@@ -60,17 +65,34 @@ exports.addOrUpdateCaseInvestigation = async (req, res) => {
         if (!investigation) {
             const investigation = new CaseInvestigation(_.pick(req.body, ["patient_id", "assigned_to"]));
             investigation._id = mongoose.Types.ObjectId();
-            // setting HCW id and date
-            investigation.notes ={
+            investigation.current_note = {
                 date : new Date(),
-                note : req.body.notes,
-                health_worker_id : req.body.loggedInUser
+                note : req.body.current_note,
             };
-            
+            investigation.notes = [];            
             await investigation.save();
             return res.status(201).send(investigation);
         }
-        investigation.set(req.body);
+        investigation.notes = req.body.notes || investigation.notes;
```

Comment:
Is this an option to update the entire investigation notes history?

Reply:
Yes, in the future if the web interface provides a means to edit the previous notes history like removing certain history notes and such